datafusion/physical-expr/src/expressions/is_false.rs

```diff
@@ -0,0 +1,143 @@
+    record_batch::RecordBatch,
+};
+use datafusion_common::DataFusionError;
+use datafusion_common::Result;
+use datafusion_common::ScalarValue;
+use datafusion_expr::ColumnarValue;
+
+/// IS FALSE expression
+#[derive(Debug)]
+pub struct IsFalseExpr {
+    /// The input expression
+    arg: Arc<dyn PhysicalExpr>,
+}
+
+impl IsFalseExpr {
+    /// Create new not expression
+    pub fn new(arg: Arc<dyn PhysicalExpr>) -> Self {
+        Self { arg }
+    }
+
+    /// Get the input expression
+    pub fn arg(&self) -> &Arc<dyn PhysicalExpr> {
+        &self.arg
+    }
+}
+
+impl std::fmt::Display for IsFalseExpr {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{} IS FALSE", self.arg)
+    }
+}
+
+impl PhysicalExpr for IsFalseExpr {
+    /// Return a reference to Any that can be used for downcasting
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn data_type(&self, _input_schema: &Schema) -> Result<DataType> {
+        Ok(DataType::Boolean)
+    }
+
+    fn nullable(&self, _input_schema: &Schema) -> Result<bool> {
+        Ok(false)
+    }
+
+    fn evaluate(&self, batch: &RecordBatch) -> Result<ColumnarValue> {
+        let arg = self.arg.evaluate(batch)?;
+        match arg {
+            ColumnarValue::Array(array) => {
+                let array_len = array.len();
+                let my_array = ColumnarValue::Array(array).into_array(array_len);
+                let true_array = BooleanArray::from(vec![Some(true)]);
+                let false_array = BooleanArray::from(vec![Some(false)]);
+                let null_array = BooleanArray::from(vec![None]);
+                let mut result_vec = vec![];
+                for i in 0..array_len {
+                    let current = (*my_array).slice(i, 1);
+                    if (*current).eq(&false_array) {
+                        result_vec.push(Some(true));
+                    } else if (*current).eq(&true_array) || (*current).eq(&null_array) {
+                        result_vec.push(Some(false));
+                    } else {
+                        return Err(DataFusionError::Execution(format!("Cannot apply 'IS FALSE' to arguments of type '<{:?}> IS FALSE'. Supported form(s): '<BOOLEAN> IS FALSE'", current.data_type())))
+                    }
+                }
+
+                let return_array = BooleanArray::from(result_vec);
+                Ok(ColumnarValue::Array(Arc::new(return_array)))
+            }
+            ColumnarValue::Scalar(scalar) => {
+                match scalar {
+                    ScalarValue::Boolean(Some(false)) => Ok(ColumnarValue::Scalar(ScalarValue::Boolean(Some(true)))),
+                    ScalarValue::Null => Ok(ColumnarValue::Scalar(ScalarValue::Boolean(Some(false)))),
+                    e => return Err(DataFusionError::Execution(format!("Cannot apply 'IS FALSE' to arguments of type '<{}> IS FALSE'. Supported form(s): '<BOOLEAN> IS FALSE'", e.get_datatype())))
+                }
```

Comment:
Thanks for making the change to return `Result`. I realize now that we could have left this in `ScalarValue` and just changed the method signature there to return `Result` as well. Moving it back to `ScalarValue` would make this logic more easily re-usable from Dask SQL. Sorry to miss that in my previous review.

Reply:
Ok cool, I've moved it back to `scalar.rs`. One quick question - would it be better to have the return type a `Result<bool>` or a `Result<Option<bool>>`? Right now I'm doing the former.

Reply:
`Result<bool>` looks appropriate to me here since there is no case that I can see where we would want to return a `None`. Thanks for making these changes.